tests/cucumber/features/step_definitions/datasets_steps.coffee

```diff
@@ -51,3 +51,27 @@ do ->
         .waitForExist('.remove')
         .click('.remove', assert.ifError)
         .call(callback)
+
+    @When 'the current user has a dataset called "$name"', (name)->
+      @browser
+        .execute((->
+          Meteor.userId()
+        ), ((err, ret) =>
+          assert.ifError(err)
+          @server.call('addDataset', {
+            name: name,
+            createdById: ret.value
+          })
+        ))
+
+    @When 'I go to the datasets page', ->
+      @browser
+        .waitForExist('.my-datasets-link')
+        .click('.my-datasets-link')
+        .waitForVisible(".my-datasets-header")
+
+    @When '"$name" should be listed under my datasets', (name)->
+      @browser
+        .pause(10000)
```

Comment:
Wait for visible didn't work? What about increasing the timeout on waitForVisible instead of the extra pause, so if the element is visible earlier it will continue?